### PR TITLE
Feature/voicemail plugin

### PR DESCRIPTION
**Pull request by LearnedVector**
voice mail plugin with audio samples for spam and a legit one generated using a tts model. the wav files are than transcribed by openai whisper.

NOTE: not the best way to store wav files on github but given this is a for demo purposes i am leaving it here.

### CODE REVIEW

File: example.py [hunk 1]
```diff
@@ -3,10 +3,13 @@
 
 from proactive_scheduler import ProactiveScheduler
 from plugins.proactive_plugin import ProactivePlugin
+from plugins.voicemail.plugin import VoiceMailPlugin
 
 USER_PROMPT = """
 I am heavily invested in bitcoin.
 Jack is heavily invested in bitcoin.
+I am a busy startup founder and often get a ton of miss phone calls.
```
Comment (kedarpujara) [inline example.py:10]: nit: add a space to keep it more readable and split between what part of the text is for what plugin

File: plugins/voicemail/plugin.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+            os.path.join(current_dir, "samples", "spam2.wav"),
+            os.path.join(current_dir, "samples", "spam3.wav"),
+            os.path.join(current_dir, "samples", "jack.wav"),
+        ]
+        self.client = get_client()
+        self.triggered = False
+
+    def invoke(self, event):
+        if random.random() < 0.05 and self.triggered == False:
+            self.triggered = True
```
Comment (kedarpujara) [inline plugins/voicemail/plugin.py:20]: When is this used in the logic? Do we need to bother to set it to true here?

**Review by kedarpujara — COMMENTED** [review]
Overall lg, just a couple nits

File: plugins/voicemail/plugin.py (hunk 2 above)
Comment (LearnedVector) [reply plugins/voicemail/plugin.py:20]: this is used so it doesn't call the api every second for example. we set it true so once it triggers, it doens't need to trigger again, so we don't waste credits on redundant transcriptions

**Pull request merged**